### PR TITLE
Refs #8756: Ensure server ca file exists before deploying.

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: manifests/katello.pp [hunk 1]
```diff
@@ -30,6 +30,7 @@
     owner  => 'root',
     group  => 'root',
     mode   => '0644',
+    File   => ["${certs::pki_dir}/certs/${certs::server_ca_name}.crt"],
```
Comment (dustints) [inline manifests/katello.pp]: @ehelms this worked for me
``` 
file { "${katello_www_pub_dir}/${certs::server_ca_name}.crt":
     ensure => link,
     target => "${certs::pki_dir}/certs/${certs::server_ca_name}.crt",                                                                     
      owner  => 'root',
      group  => 'root',
      mode   => '0644',
    } ~>
```

Comment (ehelms) [inline manifests/katello.pp]: Hrmm, I don't know that we want to symlink in this case.

Comment (dustints) [inline manifests/katello.pp]: @ehelms still got
```
 /Stage[main]/Certs::Katello/File[/var/www/html/pub/katello-server-ca.crt]: Could not evaluate: Could not retrieve information from environment production source(s) file:/etc/pki/katello/certs/katello-server-ca.crt
```
with https://gist.github.com/anonymous/067086adb9ba1a72a458

Comment (ehelms) [inline manifests/katello.pp]: Oh, shit, ha, I messed this all up. I will push what I really meant.

**Comment by ehelms** [comment]
Updated

File: manifests/katello.pp [hunk 2]
```diff
@@ -25,11 +25,12 @@
...
-    source => "${certs::pki_dir}/certs/${certs::server_ca_name}.crt",
-    owner  => 'root',
-    group  => 'root',
-    mode   => '0644',
+    ensure  => file,
+    source  => "${certs::pki_dir}/certs/${certs::server_ca_name}.crt",
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0644',
+    require => File["${certs::pki_dir}/certs/${certs::server_ca_name}.crt"],
```
Comment (dustints) [inline manifests/katello.pp]: oh right! that makes sense. ok testing it out.

**Comment by dustints** [comment]
ACK, tested: https://gist.github.com/2bdeae5173e5f911de7e
looks good!

**Pull request merged**